Introduce tag_name mode to obtain images tagged with a keyword

erAck opened the pull request:
piwigo-random-backend.php?tag_name=mybest
will obtain a random image tagged with 'mybest'. If set, tag_name
overrides a cat_id album parameter, if present.

erAck committed:
Introduce tag_name mode to obtain images tagged with a keyword

piwigo-random-backend.php?tag_name=mybest
will obtain a random image tagged with 'mybest'. If set, tag_name
overrides a cat_id album parameter, if present.

moy commented on piwigo-random-backend.php:
```diff
@@ -86,14 +93,27 @@ function check_param($param, $values) {
-  "&per_page=" . $maximages .
-  "&page=1" .
-  "&order=random";
+if ($tag_name)
```
This should be better factored. Most of the code is different in the "then" and "else" branch and should be moved out of the if/else.

erAck replied:
I find it more readable to have the entire url construction in the if/else, but ok, will do.

moy commented on piwigo-random-backend.php:
```diff
@@ -102,15 +122,20 @@ function check_param($param, $values) {
+    if ($tag_name) {
+      $page_url = (string)$image['page_url'];
+    }
+    else {
```
Please be consistent in newline placement around braces and keywords.

moy commented on piwigo-random-backend.php:
```diff
@@ -102,15 +122,20 @@ function check_param($param, $values) {
+      $page_url = (string)$image['page_url'];
+    }
+    else {
+      # Piwigo's WS returns two URLs for the image. $image['page_url'] is the URL
```
Just being curious: doesn't this statement also apply when searching by tag?

Perhaps a better test would be to check for the presence of the `categories` item in the result instead of testing `if ($tag_name) {` above.

erAck replied:
pwg.tags.getImages does not deliver categories. It's straight forward to test for $tag_name for this. Of course  $image['categories'] could be tested instead, which might be even better also for the already existing code because if it's not available or categories would be 0 the code would fail with $cats[random_int(0,-1)].
I'll change.

moy reviewed (changes requested):
Thanks for your contributions. A few nits to fix before I can merge this, but on overall this looks good.

erAck committed:
Factor out common parts of query, check categories count

As wished in
https://github.com/moy/piwigo-random/pull/5#discussion_r602886423
and suggested in
https://github.com/moy/piwigo-random/pull/5#discussion_r602886736
(also joined  } else {  ).

erAck commented:
The second commit addresses all change requests.

moy commented:
Indeed, thanks for the contribution, and the reminder :-).
